### PR TITLE
Pipeline run failure snapshot

### PR DESCRIPTION
**Pull request by montaguegabe**
Accompanies https://github.com/Parsagon/ps-scraper-web/pull/320

### CODE REVIEW

File: src/parsagon/server/server/tasks.py [hunk 1]
```diff
@@ -43,8 +57,15 @@ def run_code(run_id):
...
 
+    if error_variables is not None:
+        error_variables_serialized = {}
+        for v_name, v_value in error_variables.items():
+            error_variables_serialized[v_name] = json.dumps(v_value, default=lambda o: '<object>')
+
     error_info = {} if error is None else {
         'error': error,
         'error_transformer': error_transformer,
+        'error_page': error_page,
```
Comment (Sand1929) [inline src/parsagon/server/server/tasks.py:68]: This needs to be updated, right?

Comment (montaguegabe) [reply src/parsagon/server/server/tasks.py:68]: Yes

File: src/parsagon/server/server/tasks.py [hunk 2]
```diff
@@ -46,5 +60,6 @@ def run_code(run_id):
     error_info = {} if error is None else {
         'error': error,
         'error_transformer': error_transformer,
+        **({'error_diagnostics': {'json': error_diagnostics}} if error_diagnostics else {}),  # for debug=True only
     }
```
Comment (Sand1929) [inline src/parsagon/server/server/tasks.py:64]: Maybe it's just me, but lines 60 to 64 seem unnecessarily convoluted. I propose making this into some non-inline if statements, but I'm down to merge if you have a reason you think this is better

**Comment by montaguegabe** [comment]
Tested with changes

**Review by Sand1929 — APPROVED** [review]

**Pull request merged**